### PR TITLE
Attempt to add feature I requested in #57

### PR DESCRIPTION
**Pull request by alrahman**
#57 

I am not a C++ programmer so this should be considered a POC rather than something ready for distribution

### CODE REVIEW

**Comment by alrahman** [comment]
I had a go at adding this feature. As noted in commit message, I am not a C++ programmer so there may be mistakes. But in the testing I did, everything built and it worked as expected

**Review by stefankueng — COMMENTED** [review]
looks good. Just one request: please change the code so that syncdeleted is on by default, as it is now.
Also, you don't save/load the new setting. See CPairs::InitPairList() and CPairs::SavePairs()

File: src/PairAddDlg.cpp [hunk 1]
```diff
@@ -37,6 +37,7 @@ CPairAddDlg::CPairAddDlg(HWND hParent)
     , m_hParent(hParent)
     , m_pDropTargetOrig(nullptr)
     , m_pDropTargetCrypt(nullptr)
+    , m_syncDeleted(false)
```
Comment (stefankueng) [inline src/PairAddDlg.cpp:40]: default should be to sync deletions

File: version.txt [hunk 2]
```diff
@@ -1 +1 @@
-1.3.4.376
```
Comment (stefankueng) [inline version.txt:1]: the version.txt file is used for update checks, so please remove this change.
I'll update that file when I make a new release.

**Review by alrahman — COMMENTED** [review]
a cache file got added. should have cleaned before submitting

File: src/CryptSync.cpp [hunk 3]
```diff
@@ -88,6 +88,7 @@ int APIENTRY _tWinMain(HINSTANCE hInstance,
                              L"/use7z      : use .7z instead of .cryptsync extension\n"
                              L"/useGPG     : use GnuPG encryption\n"
                              L"/fat        : use FAT write time accuracy (2s)\n"
+                             L"/syncdeleted: sync file deletions\n"
```
[the remark this answers is not in the document]
Comment (alrahman) [reply src/CryptSync.cpp]: Done in new commit

File: src/PairAddDlg.cpp (hunk 1 above)
Comment (alrahman) [reply src/PairAddDlg.cpp]: Done in new commit

File: version.txt (hunk 2 above)
Comment (alrahman) [reply version.txt]: Change removed in new commit

File: src/CryptSync.cpp [hunk 4]
```diff
@@ -166,7 +166,7 @@ int APIENTRY _tWinMain(HINSTANCE hInstance,
         bool         use7Z        = !!parser.HasKey(L"use7z");
         bool         useGpg       = !!parser.HasKey(L"useGPG");
         bool         fat          = !!parser.HasKey(L"fat");
-        bool         syncDeleted  = !!parser.HasKey(L"syncdeleted");
+        bool         syncDeletions  = !parser.HasKey(L"nosyncdeleted");
         int          compresssize = _wtoi(parser.GetVal(L"compresssize"));
```
Comment (alrahman) [inline src/CryptSync.cpp:170]: When testing command line, Windows was throwing an exception here: Unhandled exception at 0x00ED4916 in CryptSync.exe: An invalid parameter was passed to a function that considers invalid parameters fatal.

I had to supply this parameter to complete test

**Comment by alrahman** [comment]
> Also, you don't save/load the new setting. See CPairs::InitPairList() and CPairs::SavePairs()

I missed this comment, will review and submit a new commit

**Comment by stefankueng** [comment]
looks better, but:

- you've added a cache file lzma/CS/7zip/Compress/LzmaAlone/obj/Debug/LzmaAlone.csprojAssemblyReference.cache
- in Pairs.h you initialize the member m_syncDeleted to false, should be true
-

**Comment by alrahman** [comment]
Cache file removed and Pairs.cpp updated to load & save new value

**Pull request merged**